extension/goplugin/schemas.go

```diff
@@ -714,3 +515,18 @@ func contextSetTransaction(ctx goext.Context, tx goext.ITransaction) goext.Conte
 	ctx["transaction"] = tx.RawTransaction()
 	return ctx
 }
+
+func contextGetMapResource(ctx goext.Context) map[string]interface{} {
+	return ctx["resource"].(map[string]interface{})
+}
+
+func mustGetOpenTransactionFromContext(context goext.Context) goext.ITransaction {
+	if context == nil {
+		panic("Database function called without open transaction")
```

Comment:
nit: it's better to change error message with following message.

Reply:
do you mean this should be ```return fmt.Errorf("database...")```? In my opinion this case is incorrect API usage that couldn't be spotted at compile time (at least at this moment). Because of that, I prefer early error detection and panic() is perfectly suitable for that, because it gives immediate feedback that's hard to miss.